lib/config.js

```diff
@@ -0,0 +1,286 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var fs = require('fs')
+var path = require('path')
+var url = require('url')
+var convict = require('convict')
+
+module.exports = function () {
+
+  var conf = convict({
+    env: {
+      doc: 'The current node.js environment',
+      default: 'prod',
+      format: [ 'dev', 'test', 'stage', 'prod' ],
+      env: 'NODE_ENV'
+    },
+    log: {
+      level: {
+        default: 'trace',
+        env: 'LOG_LEVEL'
+      }
+    },
+    listen: {
+      host: {
+        doc: 'The ip address the server should bind',
+        default: '127.0.0.1',
+        format: 'ipaddress',
+        env: 'IP_ADDRESS'
+      },
+      port: {
+        doc: 'The port the server should bind',
+        default: 7000,
+        format: 'port',
+        env: 'PORT'
+      }
+    },
+    allowedEmailDomains: {
+      doc: 'An array of email domains that will not be blocked or rate-limited.',
+      format: Array,
+      env: 'ALLOWED_EMAIL_DOMAINS',
+      // These are emails frequently used for testing purposes
+      default: [
+        'restmail.net',
+        'restmail.dev.lcip.org'
+      ]
+    },
+    allowedIPs: {
+      doc: 'An array of IPs that will not be blocked or rate-limited.',
+      format: Array,
+      env: 'ALLOWED_IPS',
+      default: []
+    },
+    memcache: {
+      address: {
+        doc: 'Hostname/IP:Port of the memcache server',
+        default: '127.0.0.1:11211',
+        env: 'MEMCACHE_ADDRESS'
+      },
+      recordLifetimeSeconds: {
+        doc: 'Memcache record expiry',
+        default: 900,
+        format: 'nat',
+        env: 'RECORD_LIFETIME_SECONDS'
+      }
+    },
+    bans: {
+      region: {
+        doc: 'The region where the queues live, most likely the same region we are sending email e.g. us-east-1, us-west-2, ap-southeast-2',
+        format: String,
+        default: '',
+        env: 'BANS_REGION'
+      },
+      queueUrl: {
+        doc: 'The bounce queue URL to use (should include https://sqs.<region>.amazonaws.com/<account-id>/<queue-name>)',
+        format: String,
+        default: '',
+        env: 'BANS_QUEUE_URL'
+      }
+    },
+    rules: {
+      format: Object,
+      default: {
+
+        limit_logins_15min: {
+          type: 'RateLimit',
+          action: 'checkPassword',
+          config: {
+            limit: 5,
+            period: '15 minutes'
+          }
+        },
+
+        limit_logins_daily: {
+          type: 'RateLimit',
+          action: 'checkPassword',
+          config: {
+            limit: 30,
+            period: '1 day'
+          }
+        },
```

Comment:
Here, I'm trying to re-imagine our rules as an explicit set of named rules that we can look and and understand.  They'll probably end up being slightly different to what's currently codified in code, but that seems OK as long as we're still happy they provide equivalent protection.